tests/cross_functional/kcs/test_selinux_relabel_solution.py

```diff
@@ -300,15 +304,32 @@ def test_selinux_relabel_for_existing_pvc(
         self.apply_selinux_solution_on_existing_pvc(self.pvc_obj)
 
         # Delete pod so that fix will be applied for new pod
+        POD_OBJ = ocp.OCP(
+            kind=constants.POD, namespace=config.ENV_DATA["cluster_namespace"]
+        )
+        assert modify_deploymentconfig_replica_count(
+            deploymentconfig_name=self.pod_obj.get_labels().get("name"), replica_count=0
+        ), "Failed to scale down deploymentconfig to 0"
+        POD_OBJ.wait_for_delete(resource_name=self.pod_obj.name, timeout=600)
+
+        assert modify_deploymentconfig_replica_count(
+            deploymentconfig_name=self.pod_obj.get_labels().get("name"), replica_count=1
+        ), "Failed to scale down deploymentconfig to 1"
+
         self.pod_obj = self.get_app_pod_obj()
-        self.pod_obj.delete(wait=True)
-        self.pod_obj = self.get_app_pod_obj()
-        assert wait_for_pods_to_be_running(
-            pod_names=[self.pod_obj.name], timeout=600, sleep=15
-        ), f"Pod {self.pod_obj.name} didn't reach to running state"
+
+        POD_OBJ.wait_for_resource(
```

Comment:
POD_OBJ is a single use, you can delete the line 313 and directly call ocp.OCP.wait_for_resource()